src/test/java/com/fasterxml/jackson/core/read/NumberParsingTest.java

```diff
@@ -463,6 +463,26 @@ void bigNumbers() throws Exception
         }
     }
 
+    @Test
+    void intsWith19Chars() throws Exception
+    {
+        final String[] values = new String[] {
+            "9223372036854775808", "9999999999999999999"
```

Comment:
Makes sense, could probably have based first on `Long.MAX_VALUE`? But will merge as-is.